app/src/main/java/com/vci/vectorcamapp/core/data/mappers/DeviceMapper.kt

```diff
@@ -0,0 +1,27 @@
+package com.vci.vectorcamapp.core.data.mappers
+
+import com.vci.vectorcamapp.core.data.dto.cache.DeviceCacheDto
+import com.vci.vectorcamapp.core.domain.model.Device
+
+fun DeviceCacheDto.toDomain(): Device? {
+    return if (programId == -1) {
+        null
+    } else {
+        Device(
+            id = id,
+            model = model,
+            registeredAt = registeredAt,
+            submittedAt = submittedAt
+        )
+    }
+}
+
+fun Device.toDto(programId: Int): DeviceCacheDto {
+    return DeviceCacheDto(
+        id = id,
+        programId = programId,
+        model = model,
+        registeredAt = registeredAt,
+        submittedAt = submittedAt
+    )
+}
```

Comment:
I like the thought process, but the Device.toDto function is not clear. There are 2 different types of DTOs for the device, DeviceCacheDto, DeviceDto, PostDeviceRequestDto, PostDeviceResponseDto. By saying toDto whenever you use this function, I would not know which Dto you are trying to convert to. I would think you are converting the Device to DeviceDto.